server/test/test_latest_mlp_model_errors.py

```
@@ -0,0 +1,93 @@
+        f"http://localhost:{PORT}/model-version", headers=headers
+    )
+    while True:
+        if proc.poll() is not None:
+            raise RuntimeError("server failed to start")
+        try:
+            with urllib.request.urlopen(req) as resp:
+                if resp.getcode() == 200:
+                    break
+        except Exception:
+            if time.time() - start > 30:
+                raise RuntimeError("server did not start in time")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)